undefined reference when enabling CONFIG_STACK_CANARIES

Add:
```
CONFIG_STACK_CANARIES=y
```

to your choice of application (e.g. `hello_world`) and platform (e.g. `nrf52840_pca10056`, `mimxrt1060_evk`, `hifive1_revb`) and builds will fail:
```
Memory region         Used Size  Region Size  %age Used
           FLASH:       14736 B         1 MB      1.41%
            SRAM:        3956 B       256 KB      1.51%
        IDT_LIST:          56 B         2 KB      2.73/usr/local/zephyr-sdk-0.10.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/8.3.0/../../../../arm-zephyr-eabi/bin/ld: zephyr/kernel/libkernel.a(init.c.obj): in function `z_early_boot_rand_get':
/mnt/nordic/zp/zephyr/kernel/init.c:469: undefined reference to `sys_rand32_get'
collect2: error: ld returned 1 exit status
```

--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -620,6 +620,7 @@ menu "Security Options"
 
 config STACK_CANARIES
 	bool "Compiler stack canaries"
+	depends on ENTROPY_GENERATOR || TEST_RANDOM_GENERATOR
 	help
 	  This option enables compiler stack canaries.
 

